### PR TITLE
feat: add Status

### PR DESCRIPTION
**Pull request by charlesnguyen**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**
The Status component displays a label with icon in function of the status and when the mouse is over the label, the component displays a set of actions.
This component will be used in a collapsible Panel. 
See http://guidelines.talend.com/document/135457#/export-history

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: src/Status/Status.component.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+						onClick: action('onDelete'),
+					}],
+			onMouseEnter: action('onMouseEnter'),
+			onMouseLeave: action('onMouseLeave'),
+
+	};
+ 	<Status {...status}/>
+ */
+
+const STATUS_IN_PROGRESS = 'inProgress';
```
Comment (frassinier) [inline src/Status/Status.component.js]: You should export this constants

File: src/Status/Status.component.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+function renderIcon(status, icon) {
+	if (status === STATUS_IN_PROGRESS) {
+		return ( icon ? <Icon name={icon}/> : <CircularProgress size={'small'}/> );
+	}
+	return ( icon && <Icon name={icon}/> );
+}
+
+function renderStatus(isMouseOver, label, actionsOnMouseOver) {
+	if (isMouseOver) {
```
Comment (frassinier) [inline src/Status/Status.component.js]: IMO, you should add actions (buttons) depending of the current status. 
Then let button be there and with CSS add a `visibility: hidden` and on _:focus_ or on _:hover_ reset to `visibility: visible`

File: src/Status/Status.component.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+		icon,
+		isMouseOver,
+		actionsOnMouseOver,
+		onMouseEnter,
+		onMouseLeave,
+		} = props;
+
+	return (
+		<div className={`text-${getbsStyleFromStatus(status)}`}
+			onMouseEnter={(event) => onMouseEnter (event, {isMouseOver: true})}
```
Comment (frassinier) [inline src/Status/Status.component.js]: Overkill! Let's deal with CSS!

File: src/Status/Status.component.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+		</div>
+	);
+}
+
+Status.propTypes = {
+	status: PropTypes.oneOf([STATUS_IN_PROGRESS, STATUS_SUCCESSFUL, STATUS_FAILED, STATUS_CANCELED]),
+	label: PropTypes.string.isRequired,
+	icon: PropTypes.string,
+	isMouseOver: PropTypes.bool,
+	actionsOnMouseOver: PropTypes.shape(Actions.propTypes.actions),
```
Comment (frassinier) [inline src/Status/Status.component.js]: Rename it to `actions`

File: src/Status/Status.component.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+			{renderStatus(isMouseOver, label, actionsOnMouseOver)}
+		</div>
+	);
+}
+
+Status.propTypes = {
+	status: PropTypes.oneOf([STATUS_IN_PROGRESS, STATUS_SUCCESSFUL, STATUS_FAILED, STATUS_CANCELED]),
+	label: PropTypes.string.isRequired,
+	icon: PropTypes.string,
+	isMouseOver: PropTypes.bool,
```
Comment (frassinier) [inline src/Status/Status.component.js]: Delete it

File: src/Status/Status.component.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+	actionsOnMouseOver: PropTypes.shape(Actions.propTypes.actions),
+	onMouseEnter: PropTypes.func.isRequired,
```
Comment (frassinier) [inline src/Status/Status.component.js]: Delete it

File: src/Status/Status.component.js [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+	onMouseLeave: PropTypes.func.isRequired,
```
Comment (frassinier) [inline src/Status/Status.component.js]: Delete it

File: src/Status/Status.snap.test.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+			label: 'cancel',
+			icon: 'fa fa-cancel',
+			onClick: jest.fn(),
+			bsSize: 'small',
+		},
+		{
+			label: 'delete',
+			icon: 'fa fa-delete',
+			onClick: jest.fn(),
+			bsSize: 'small',
```
Comment (frassinier) [inline src/Status/Status.snap.test.js]: Do we need this?

Comment (charlesnguyen) [inline src/Status/Status.snap.test.js]: To set action buttons size.
Otherwise, we set it 'small' by default.

Comment (frassinier) [inline src/Status/Status.snap.test.js]: Good option. `small` is a good default.

Comment (charlesnguyen) [inline src/Status/Status.snap.test.js]: I just looked at ActionComponent, bsStyle: 'default' by default. So we let it at 'default' though?

**Pull request merged**